packages/gatsby/src/services/initialize.ts

```diff
@@ -217,7 +216,9 @@ export async function initialize({
 
     //  track usage of feature
     enabledConfigFlags.forEach(flag => {
-      telemetry.trackFeatureIsUsed(flag.telemetryId)
+      if (flag.telemetryId !== DO_NOT_TRACK_THIS_FLAG) {
```

Comment:
If `telemetryId` is optional (as suggested above):
```suggestion
      if (flag.telemetryId) {
```

Reply:
Ended up using exactly that in https://github.com/gatsbyjs/gatsby/pull/28787/commits/e22fc5e9a42e898fb509d6227a61c410d86cfb78 ;)